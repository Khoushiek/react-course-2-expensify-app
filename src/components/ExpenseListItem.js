import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";

const numberFormat = (value) =>
  new Intl.NumberFormat('en-IN', {
    style: 'currency',
    currency: 'INR'
  }).format(value);

export default (props) => (
  <div>
    <Link to={`/edit/${props.id}`}>
      <h3>{props.description}</h3>
    </Link>
    <p>
      {numberFormat(props.amount / 100 )}
      &nbsp; - &nbsp;  
      {moment(props.createdAt).format("MMMM Do YYYY")}</p>
  </div>
);
