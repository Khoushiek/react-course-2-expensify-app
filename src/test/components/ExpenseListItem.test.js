import {shallow} from "enzyme";
import expenses from "../fixtures/expenses";
import React from "react";
import ExpenseListItem from "../../components/ExpenseListItem";

test("should return Expense List Item", () => {
  const expense = expenses[1]
  const wrapper = shallow(<ExpenseListItem 
    description = {expense.description} 
    amount={expense.amount} 
    createdAt={expense.createdAt}/>);
  expect(wrapper).toMatchSnapshot();
})