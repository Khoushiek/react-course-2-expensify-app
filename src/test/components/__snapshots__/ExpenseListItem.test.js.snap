// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return Expense List Item 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/undefined"
  >
    <h3>
      Gum
    </h3>
  </Link>
  <p>
    ₹2.50
      -  
    January 5th 1970
  </p>
</div>
`;
